hazelcast/src/main/java/com/hazelcast/config/WanPublisherConfig.java

```diff
@@ -82,54 +90,85 @@ public int getQueueCapacity() {
     }
 
     /**
-     * Set the capacity of the queue for WAN replication events. IMap, ICache, normal and backup events count against
-     * the queue capacity separately. When the queue capacity is reached, backup events are dropped while normal
+     * Sets the capacity of the queue for WAN replication events. IMap, ICache,
+     * normal and backup events count against the queue capacity separately.
+     * When the queue capacity is reached, backup events are dropped while normal
```

Comment:
Our default implementation is not fully in sync with this.
The check that decides whether or not to drop a backup event takes only the backup event count into consideration.

Similarly, we think the queue is full only if the primary counter reaches the threshold. With the default implementation, we currently allow to keep events in memory up to `2*queueCapacity`.